### PR TITLE
TSDK-733 TSDK-738 Confirm redemption

### PR DESCRIPTION
**Pull request by mundacho**
## Purpose

The goal of this PR is to implement the Confirm redemption WS.

## Approach

We implemented the WS, added parameters to the bridge and tested it. We also added session management (in memory) and wallet management.

## Testing

We created unit tests for the different controllers and managers. We also run it locally and tested that the transactions work.

## Tickets

* TSDK-733
* TSDK-738

### CODE REVIEW

File: topl-btc-bridge/src/main/scala/co/topl/bridge/controllers/StartSessionController.scala [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      pUserPKey: String,
+      bridgePKey: String,
+      blockToRecover: Int,
+      btcNetwork: BitcoinNetworkIdentifiers
+  ): F[SessionInfo] = {
+    import cats.implicits._
+    for {
+      hash <- Sync[F].fromOption(
+        ByteVector.fromHex(sha256),
+        InvalidHash(s"Invalid hash $sha256")
```
Comment (DiademShoukralla) [inline topl-btc-bridge/src/main/scala/co/topl/bridge/controllers/StartSessionController.scala:39]: Perhaps we can also verify that the hash contains 32 bytes

**Review by DiademShoukralla — APPROVED** [review]
LGTM! I just had a few comments but nothing to block merging

File: shared/src/main/scala/co/topl/shared/utils/KeyGenerationUtils.scala [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+  def signWithKeyManager[F[_]: Sync](
+      km: BIP39KeyManager,
+      txBytes: ByteVector,
+      currentIdx: Int
+  ): F[String] = {
+    import cats.implicits._
+    for {
+      signed <- Sync[F].delay(
+        km.toSign(HDPath.fromString("m/84'/1'/0'/0/" + currentIdx))
```
[the remark this answers is not in the document]
Comment (mundacho) [reply shared/src/main/scala/co/topl/shared/utils/KeyGenerationUtils.scala:28]: We can discuss this in the checkin. 👍

File: shared/src/main/scala/co/topl/shared/InputOutput.scala [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+case class StartSessionRequest(
+    pkey: String,
+    sha256: String
+)
+
+case class StartSessionResponse(
+    sessionID: String,
+    script: String,
+    escrowAddress: String,
```
[the remark this answers is not in the document]
Comment (mundacho) [reply shared/src/main/scala/co/topl/shared/InputOutput.scala:11]: yes, this is a bitcoin address. Concerning the topl address, probably yes, but this version we are not yet doing pegout 😬

File: shared/src/main/scala/co/topl/shared/InputOutput.scala [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    descriptor: String
+)
+
+case class ConfirmRedemptionRequest(
+    sessionID: String,
+    inputTxId: String,
+    inputIndex: Int,
+    feePerByte: Int,
+    amount: Long,
+    secret: String
```
[the remark this answers is not in the document]
Comment (mundacho) [reply shared/src/main/scala/co/topl/shared/InputOutput.scala:21]: yes, probably the whole ws will be removed when the network monitoring and secret extraction is completed.

File: topl-btc-bridge/src/main/scala/co/topl/bridge/controllers/StartSessionController.scala (hunk 1 above)
Comment (mundacho) [reply topl-btc-bridge/src/main/scala/co/topl/bridge/controllers/StartSessionController.scala:39]: I'll add this in my next task.

**Pull request merged**